float32[] trajectory
int32 num_repeats
---
